drivers/FileHandle.h

```diff
@@ -100,43 +124,20 @@ class FileHandle {
      *    0 on success or un-needed,
      *   -1 on error
      */
-    virtual int fsync() = 0;
-
-    virtual off_t flen() {
-        lock();
-        /* remember our current position */
-        off_t pos = lseek(0, SEEK_CUR);
-        if(pos == -1) {
-            unlock();
-            return -1;
-        }
-        /* seek to the end to get the file length */
-        off_t res = lseek(0, SEEK_END);
-        /* return to our old position */
-        lseek(pos, SEEK_SET);
-        unlock();
-        return res;
-    }
-
-    virtual ~FileHandle() {};
-
-protected:
-
-    /** Acquire exclusive access to this object.
-     */
-    virtual void lock() {
-        // Stub
-    }
+    MBED_DEPRECATED_SINCE("mbed-os-5.4", "Replaced by FileLike::sync")
+    virtual int fsync() { return sync(); }
```

Comment:
Well, no, they weren't used. retarget didn't call them, nor did anyone else. They were kind of just like a placeholder for a lock that may or may not be used internally.

So I think they can go.

There is another conceptual thought on the subject though - some FileHandles may be usable from interrupt (and hence directly from attach callback). Their internal lock could be a critical section, rather than a mutex. I believe our BufferedSerial could maybe be like this. I'm wondering whether this could be an API feature that could ever be relied on.

Could it maybe be the case that having the lock visible might make that overrideable, like it is for some of the HAL devices? Override lock to prevent a mutex claim?

My feeling is that I don't want to do that - not abstract enough. Instead, I suggest internal lock behaviour is up to the class. Generic callers of FileHandle methods should assume that by default it has to be called from foreground context, but a given FileHandle might document that it is usable from interrupt.